pkg/reconciler/apis/apiexport/apiexport_controller.go

```diff
@@ -205,19 +204,13 @@ func (c *controller) enqueueAPIExport(obj interface{}) {
 }
 
 func (c *controller) enqueueAllAPIExports(clusterWorkspaceShard interface{}) {
-	clusterWorkspaceShardKey, err := cache.DeletionHandlingMetaNamespaceKeyFunc(clusterWorkspaceShard)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
 	list, err := c.apiExportLister.List(labels.Everything())
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
 
-	logger := logging.WithReconciler(klog.Background(), controllerName).WithValues("ClusterWorkspacShard", clusterWorkspaceShardKey)
+	logger := logging.WithObject(logging.WithReconciler(klog.Background(), controllerName), clusterWorkspaceShard.(*tenancyv1alpha1.ClusterWorkspaceShard))
```

Comment:
(no action required) I'd split this in 2 to make it easier to read

Reply:
Fairly pervasive at this point - perhaps I can try to make this a builder pattern in a follow-up and fix them all?

Reply:
SGTM